From 3cbd8e6795a466b6f549b1073c2b4e15a4cffe6b Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 10 Oct 2022 07:43:29 +0300
Subject: [PATCH] [SW-0] gaudi3: onix: remove empty routine for debug rr

This patch remove empty functions for debug rr routine on onix.
The debug RR configuration doesn't hang or cause ARC exception on onix
and there wont be any functionality impact due to this RR configuration.

Change-Id: I77910c61feb4df47ab4b32f937184783fcfb834d
---

diff --git a/arch/gaudi3/arc/onix.c b/arch/gaudi3/arc/onix.c
index 883e357..c46b2c0 100644
--- a/arch/gaudi3/arc/onix.c
+++ b/arch/gaudi3/arc/onix.c
@@ -10,8 +10,6 @@
 #define SYS_CLOCK_FREQUENCY	35000000ULL
 
 void security_rr_close_pcie_config(void) {}
-void security_rr_config_debug_if_lbw(__unused struct rom_rr_ctx *rr_ctx) {}
-void security_rr_config_debug_if_hbw(__unused struct rom_rr_ctx *rr_ctx) {}
 void rom_pcie_recovery(void) {}
 
 uint64_t arcboot_rom_get_timestamp_freq(void)
diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index 0a11fc8..0ea38eb 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -777,7 +777,7 @@
 }
 
 #define DEBUG_IF_ENABLE_VAL		0
-__weak void security_rr_config_debug_if_lbw(struct rom_rr_ctx *rr_ctx)
+void security_rr_config_debug_if_lbw(struct rom_rr_ctx *rr_ctx)
 {
 	if ((arcboot_rom_global.config_fuse &
 		EFUSE_SYSCFG_DISABLE_JTAG_IF_MASK) == DEBUG_IF_ENABLE_VAL)
@@ -788,7 +788,7 @@
 		security_rr_twi_lbw_config(rr_ctx);
 }
 
-__weak void security_rr_config_debug_if_hbw(struct rom_rr_ctx *rr_ctx)
+void security_rr_config_debug_if_hbw(struct rom_rr_ctx *rr_ctx)
 {
 	if ((arcboot_rom_global.config_fuse &
 		EFUSE_SYSCFG_DISABLE_JTAG_IF_MASK) == DEBUG_IF_ENABLE_VAL)
